buildlib/azure-pipelines-perf.yml

```diff
@@ -106,11 +109,43 @@ stages:
                 --after "$SHA_After" \
                 --module hpcx-gcc \
                 --filter \
-                --extra-ompi-opts " -H $(perf_nodes)" | tee $(WorkDir)/results.txt
-            displayName: Perf test
+                --omb-cuda \
+                --extra-ompi-opts " -H $(perf_nodes)" | tee $(WorkDir)/results-multinode.txt
+            displayName: Perf test - multi node
             workingDirectory: $(WorkDir)
 
-          - script: $(WorkDir)/ucx/buildlib/tools/perf_results.py $(WorkDir)/results.txt $(threshold)
+          - bash: |
+              set -eE
+              SHA_After=$(echo $(Build.SourceVersionMessage) | awk '{print $2}')
+              SHA_Before=$(echo $(Build.SourceVersionMessage) | awk '{print $4}')
+
+              python3 -m venv VENV
+              source VENV/bin/activate
+              cd $(WorkDir)/PerfX
+              module load /hpc/local/etc/modulefiles/hpcx-gcc
+
+              set -xeE -o pipefail
+              # Pick random node
+              IFS=',' read -ra nodes <<< "$(perf_nodes)"
+              node=${nodes[RANDOM % 2]}
+
+              python3 -u perfx.py \
+                --skip-update \
+                --skip-build \
+                --use-cache \
+                --source-branch $(Build.SourceBranch) \
+                --config ucx-rdmz.yml \
+                --before "$SHA_Before" \
+                --after "$SHA_After" \
+                --module hpcx-gcc \
+                --filter \
+                --omb-cuda \
+                --setup rdmz-intra \
+                --extra-ompi-opts " -H $node,$node" | tee $(WorkDir)/results-singlenode.txt
+            displayName: Perf test - single node
+            workingDirectory: $(WorkDir)
```

Comment:
Not much in common between the two steps:
perfx.py flags differ significantly, modules, venv init - not the same all of them.

Reply:
Do I understand correctly that `--skip-build` flag is set because build happens during the first perfX execution?

Is it possible to create some "Initialization" step to download requirements and perform build to make the code for multi and single node mode common?

Also, is it possible to add CUDA tests for single-node too?

Reply:
Perfx also has `--skip-run` :)
So should be possible to introduce such initialisation.

Reply:
`--skip-run` didn't go well when I tried it, but we can further debug it.

All tests get run in both multi-node and single-node modes.

Reply:
Then it's strange that we do load `hpcx-gcc` module for both single and multi node runs, but `dev/cuda-latest` only for multi.

Reply:
`hpcx-gcc` module provides OMPI.
CUDA module is only needed for the build phase.